packages/@react-aria/menu/src/useMenuTrigger.ts

```diff
@@ -34,52 +36,89 @@ interface MenuTriggerAria {
  * @param props - Props for the menu trigger.
  * @param state - State for the menu trigger.
  */
-export function useMenuTrigger(props: MenuTriggerAriaProps, state: MenuTriggerState, ref: RefObject<HTMLElement>): MenuTriggerAria {
+export function useMenuTrigger(props: MenuTriggerAriaProps, state: MenuTriggerState, ref: RefObject<HTMLElement>, trigger?: MenuTriggerType): MenuTriggerAria {
   let {
     type = 'menu' as MenuTriggerAriaProps['type']
   } = props;
 
   let menuTriggerId = useId();
   let {triggerProps, overlayProps} = useOverlayTrigger({type}, state, ref);
+  
+  const handleArrowKeyBehaviour = (e) => {
+    switch (e.key) {
+      case 'ArrowDown':
+        e.preventDefault();
+        state.toggle('first');
+        break;
+      case 'ArrowUp':
+        e.preventDefault();
+        state.toggle('last');
+        break;
+    }
+  };
 
-  let onKeyDown = (e) => {
+  const handleLongPressAltBehaviour = (e) => {
+    if (e.altKey) {
+      handleArrowKeyBehaviour(e);
+    }
+  };
+  
+  const onKeyDown = (e) => {
     if ((typeof e.isDefaultPrevented === 'function' && e.isDefaultPrevented()) || e.defaultPrevented) {
       return;
     }
 
     if (ref && ref.current) {
-      switch (e.key) {
-        case 'ArrowDown':
-          e.preventDefault();
-          state.toggle('first');
-          break;
-        case 'ArrowUp':
-          e.preventDefault();
-          state.toggle('last');
-          break;
+      if (trigger === 'longPress') {
+        handleLongPressAltBehaviour(e);
+      } else  {
+        handleArrowKeyBehaviour(e);
       }
     }
   };
 
-  return {
-    menuTriggerProps: {
-      ...triggerProps,
-      id: menuTriggerId,
-      onPressStart(e) {
-        // For consistency with native, open the menu on mouse/key down, but touch up.
-        if (e.pointerType !== 'touch') {
-          // If opened with a keyboard or screen reader, auto focus the first item.
-          // Otherwise, the menu itself will be focused.
-          state.toggle(e.pointerType === 'keyboard' || e.pointerType === 'virtual' ? 'first' : null);
-        }
-      },
-      onPress(e) {
-        if (e.pointerType === 'touch') {
-          state.toggle();
-        }
-      },
-      onKeyDown
+  const longPressProps = useLongPress({
+    onLongPress(e) {
```

Comment:
If you make the change above and only allow mouse and touch pointer types for long presses, you could always call `state.toggle('first')` here.